napari/_qt/containers/_layer_delegate.py

```diff
@@ -112,6 +124,39 @@ def get_layer_icon(
         option.decorationPosition = option.Right  # put icon on the right
         option.features |= option.HasDecoration
 
+    def _check_loaded(self):
+        """
+        Check loading state of the layer and pause loading movie if necessary.
+        """
+        if self.layer_index and self.layer_index.isValid():
+            if self.layer_index.data(LoadedRole):
+                self.load_movie.setPaused(True)
+                self.loading_frame_changed.emit()
+
+    def _paint_loading(
+        self,
+        painter: QPainter,
+        option: QStyleOptionViewItem,
+        index: QtCore.QModelIndex,
+    ):
```

Comment:
If I understand correctly, this function is called whenever paint events occur. So if the layer is loaded, every paint event will start a single shot timer. It should be started when movie is already running. 

Also, this looks like something that will be a reason for tests segfaults.

Reply:
Thanks for the feedback @Czaki ! You are correct, this function is called whenever a paint event occur. Regarding when to to start the single shot timer, maybe this definition of the `_paint_loading method` could be better?:

```python
   def _paint_loading(
        self,
        painter: QPainter,
        option: QStyleOptionViewItem,
        index: QtCore.QModelIndex,
    ):
        """Paint loading layer indicator."""
        loaded = index.data(LoadedRole)
        load_rect = option.rect.translated(55, 8)
        h = index.data(Qt.ItemDataRole.SizeHintRole).height() - 16
        load_rect.setWidth(h)
        load_rect.setHeight(h)

        if loaded and self.load_movie.state() == QMovie.Running:
            # Add some delay to check if the layer is still loaded to
            # prevent blinking from the loading indicator
            # when pausing the movie.
            self.layer_index = index
            QTimer.singleShot(100, self._check_loaded)
        elif not loaded:
            self.load_movie.start()

        if self.load_movie.state() == QMovie.Running:
            painter.drawPixmap(load_rect, self.load_movie.currentPixmap())
```

Regarding segfaults, you are right, in fact [I had a test segfaulting before since initialy I used a lambda to call `self._check_loaded` passing to the method the `index` variable](https://github.com/napari/napari/actions/runs/3515853535/jobs/5891642145#step:8:216). However, changing that approach to instead set the `index` as a class attribute (to prevent it from being garbage collected) seems like helped. If you have any further recommendations to be sure we prevent segfaults due to the `QTimer` usage here let me know!

Reply:
I think the latest changes pushed here should prevent segfaults. Quite handy to have a test that checks for timers :+1:

Reply:
Seems like the test are passing now 🎉 (although I had use `QObject.timerEvent` instead of `QTimer.singleShot`)